astroquery/esasky/core.py

```diff
@@ -0,0 +1,624 @@
+        return None  
+
+    def _find_observation_tap_table_name(self, mission_name, cache):
+        return self._find_mission_tap_table_name(self._fetch_and_parse_json(self.__OBSERVATIONS_STRING, cache), mission_name)
+
+    def _find_catalog_tap_table_name(self, mission_name, cache):
+        return self._find_mission_tap_table_name(self._fetch_and_parse_json(self.__CATALOGS_STRING, cache), mission_name)
+    
+    def _find_mission_tap_table_name(self, json, mission_name):
+        for i in range(len(json)):
+            if (json[i][self.__MISSION_STRING].lower() == mission_name.lower()):
+                return json[i][self.__TAP_TABLE_STRING]
+        
+        raise ValueError("Input %s not available." %mission_name)
+        return None  
+  
+    def _get_observation_json(self, cache):
+        return self._fetch_and_parse_json(self.__OBSERVATIONS_STRING, cache)
+
+    def _get_catalogs_json(self, cache):
+        return self._fetch_and_parse_json(self.__CATALOGS_STRING, cache)
+  
+    def _fetch_and_parse_json(self, object_name, cache):
+        url = self.URLbase + "/" + object_name
+        response = self._request('GET', url, cache = cache)
+        string_response = response.content.decode('utf-8')
+        json_response = json.loads(string_response)
+        return json_response[object_name]
+    
+    def _json_object_field_to_list(self, json, field_name):
+        response_list = []
+        for i in range(len(json)):
+            response_list.append(json[i][field_name])
+        return response_list
+    
+    def _create_request_payload(self, query):
+        return {'REQUEST':'doQuery', 'LANG':'ADQL', 'FORMAT': 'VOTABLE', 'QUERY': query}
+    
+    def _get_and_parse_from_tap(self, request_payload, cache):
+        response = self._send_get_request("/tap/sync", request_payload, cache)
+        return self._parse_xml_table(response)
+    
+    def _send_get_request(self, url_extension, request_payload, cache):
+        URL = self.URLbase + url_extension
+        return self._request('GET', URL, params=request_payload,
+                                 timeout=self.TIMEOUT, cache=cache)
+
+    def _parse_xml_table(self, response):
+        # try to parse the result into an astropy.Table, else
+        # return the raw result with an informative error message.
+        try:
+            tf = six.BytesIO(response.content)
+            vo_table = votable.parse(tf, pedantic = False)
+            first_table = vo_table.get_first_table()
+            table = first_table.to_table(use_names_over_ids = True)
+            return table
+        except Exception as ex:
+            self.response = response
+            self.table_parse_error = ex
+            raise TableParseError(
+                "Failed to parse ESASky VOTABLE result! The raw response can be "
+                "found in self.response, and the error in "
+                "self.table_parse_error.")
+        return Table()
```

Comment:
should this return statement be here?  I think the above code will either return successfully or raise an exception